opensearch/services/datatypes/datatypes.php

```diff
@@ -0,0 +1,430 @@
+	* @return string query footer component (e.g. 'GROUP BY ?label ?id')
+	*/
+	public function getQueryFooter($type, $limit=null, $offset=0, $sort=null) {
+	
+		$footer = "";
+		switch($type) {
+			case "datatypes":
+				$footer .= " GROUP BY ?datatype ?dco_id ?label ?year ?parameterLabel ?access";
+				$footer .= " ORDER BY ?label";
+				if ($limit)	$footer .= " LIMIT $limit OFFSET $offset";
+				break;
+			case "count":
+				break;
+			default:
+				$footer .= " GROUP BY ?label ?id";
+				break;
+		}
+		return $footer;
+	}
+	
+	/**
+	  * Return SPARQL query WHERE clause minus constraint clauses for specified search type
+	  * @param string $type search type (e.g. 'datasets', 'authors', 'keywords')
+	  * @return string WHERE clause component minus constraint clauses (e.g. '?dataset a dcat:Dataset . ')
+	  */
+	public function getQueryBody($type) {
+		
+		$body = "";
+		switch($type) {
+			case "creationYear":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype  dco:createdAtTime ?t . ";
+				$body .= "BIND(str(?t) AS ?year) . ";
+				break;
+				
+			case "creator":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype prov:qualifiedAttribution [prov:hadRole ?r ] . ";
+				$body .= "?r a prov:Role . ";
+				$body .= "?c rdf:type ?r . ";
+				$body .= "?c a dco:creator . ";
+				$body .= "BIND(str(?c) AS ?creator) . ";
+				break;
+
+			case "parameter":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype dco:hasParameter [rdfs:label ?l] . ";
+				$body .= "BIND(str(?l) AS ?parameterLabel) . ";
+				break;
+
+			case "sourceStandard":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype dco:sourceStandard [rdfs:label ?l] . ";
+				$body .= "BIND(str(?l) AS ?sourceStandard_label) . ";
+				break;
+
+			case "subjectArea":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype vivo:hasSubjectArea ?s . ";
+				$body .= "?s a skos:Concept . ";
+				break;
+
+			case "datatypes":
+				$body .= "?datatype a dco:DataType . ";
+				$body .= "?datatype rdfs:label ?l . ";
+				$body .= "OPTIONAL { ?datatype dco:hasDcoId ?id . } ";
+				$body .= "OPTIONAL { ?datatype dco:createdAtTime ?ct . } ";
+				$body .= "OPTIONAL { ?datatype dco:lastModifiedAtTIme ?mt . } ";
+				$body .= "OPTIONAL { ?datatype dco:sourceStandard ?source . ?source rdfs:label ?sl . } ";
+				$body .= "OPTIONAL { ?datatype dco:expectedUses ?use . } ";
+				$body .= "OPTIONAL { ?datatype dco:hasParameter ?param . ?param rdfs:label ?pl . } ";
+				$body .= "OPTIONAL { ?datatype dco:sourceDataType ?s . ?s rdfs:label ?sourceLabel . } ";
+				$body .= "BIND(str(?l) AS ?label) . ";
+				$body .= "BIND(str(?id) AS ?dco_id) . ";
+				$body .= "BIND(str(?ct) AS ?createdTime) . ";
+				$body .= "BIND(str(?mt) AS ?modifiedTime) . ";
+				$body .= "BIND(str(?sl) AS ?sourceStandard_label) . ";
+				$body .= "BIND(str(?use) AS ?use) . ";
+				$body .= "BIND(str(?pl) AS ?parameterLabel) . ";
```

Comment:
Yes, there is one sample data type instance in UDCO that has multiple parameters.